lib/bulkrax.rb

```diff
@@ -98,7 +98,22 @@ def collection_model_class
     attr_writer :collection_model_class
 
     def collection_model_internal_resource
-      collection_model_class.try(:internal_resource) || collection_model_class.to_s
+      # WARN: Using #try on :internal_resource can yield unexpected results.
+      # If the method is undefined, it can return a truthy value instead of
+      # the typical nil.
+      #
+      # E.g.
+      # ```ruby
+      # Hyrax::FileSet.try(:internal_resource) || 'hi'
+      # => #<Dry::Types::Result::Failure input=:internal_resource error=...
+      # ```
+      ir = begin
+             collection_model_class.internal_resource
+           rescue NoMethodError
+             nil
+           end
+
+      ir.presence || collection_model_class.to_s
```

Comment:
```suggestion
      # WARN: Using #try on :internal_resource can yield unexpected results.
      # If the method is undefined, it can return a truthy value instead of
      # the typical nil.
      #
      # E.g.
      # ```ruby
      # Hyrax::FileSet.try(:internal_resource) || 'hi'
      # => #<Dry::Types::Result::Failure input=:internal_resource error=...
      # ```
      ir = collection_model_class.respond_to?(:internal_resource) ?
              collection_model_class.internal_resource : collection_model_class.to_s
```

Reply:
```ruby
irb(main):001:0> Hyrax::FileSet.try(:internal_resource) || 'hi'
=> #<Dry::Types::Result::Failure input=:internal_resource error=#<Dry::Struct::Error: [Hyrax::FileSet.new] can't convert Symbol into Hash>>
irb(main):002:0> Hyrax::FileSet.internal_resoruce
(irb):2:in `<main>': undefined method `internal_resoruce' for Hyrax::FileSet:Class (NoMethodError)
irb(main):003:0> Hyrax::FileSet.respond_to?(:internal_resoruce)
=> false
```

Reply:
Thanks, that's much cleaner 🙂